### ISSUE TITLE
Add build scripts for Linux

### ISSUE THREAD

**Issue by SasLuca** [SasLuca/rayfork#11]
Currently, there is a simple CMake file used to build the examples and a batch script that invokes the `cl` msvc compiler from the command line.

We need to add support for more compilers in the build scripts (eg: gcc and clang) and maybe support to build the examples with more build systems (eg: Make, premake, gradle).

**Comment by ChrisDill** [comment]
I made a build.sh script based on the included [build.bat](https://github.com/SasLuca/rayfork/blob/master/build.bat) so I can build rayfork on linux.

I am able to build and run basic_shapes, collision_area and geometric_shapes. I based the script on other raylib tests so some parts might not be needed. Feel free to change it.

```bash
#!/usr/bin/bash

FLAGS="-g -Wall -Wno-missing-braces -D_DEFAULT_SOURCE"
INCLUDES="-I../../rayfork -I../../dependencies -I../../examples/geometric_shapes/ -I../../examples/geometric_shapes/dependencies"
LIBRARIES="-lGL -lm -lpthread -ldl -lrt -lX11"

mkdir -p bin
pushd bin

mkdir -p basic_shapes
pushd basic_shapes
gcc -o basic_shapes ../../examples/basic_shapes/main.c ../../examples/basic_shapes/game.c ../../examples/basic_shapes/rayfork_renderer.c ../../examples/basic_shapes/dependencies/glad/glad.c $FLAGS $INCLUDES $LIBRARIES;
popd

mkdir -p collision_area
pushd collision_area
gcc -o collision_area ../../examples/collision_area/main.c ../../examples/collision_area/game.c ../../examples/collision_area/rayfork_renderer.c ../../examples/collision_area/dependencies/glad/glad.c $FLAGS $INCLUDES $LIBRARIES;
popd

mkdir -p geometric_shapes
pushd geometric_shapes
gcc -o geometric_shapes ../../examples/geometric_shapes/main.c ../../examples/geometric_shapes/game.c ../../examples/geometric_shapes/rayfork_renderer.c ../../examples/geometric_shapes/dependencies/glad/glad.c $FLAGS $INCLUDES $LIBRARIES;
popd

popd
```

# Rayfork example running on linux:
![image](https://user-images.githubusercontent.com/7503113/71185361-3391d480-2273-11ea-821c-5bf164f206f4.png)

Also are the build scripts intended to build all ported examples or should it be left for other build systems instead?

**Comment by SasLuca** [comment]
Thanks a lot for this!

I don't have a solid policy regarding the build scripts, I just want it to be as simple and convenient for users to compile the examples and run them. Since the examples are small I don't see a problem with one script compiling all of them. If a user wants to compile just one specific example for some reason they can easily modify the build scripts.

**Comment by SasLuca** [comment]
Added the fix: 42066fdcf877122fe88a2d2c0cde72a9bac26608
